### PR TITLE
All hail The Pickle Jar, harbringer of better crafting

### LINKED ISSUE #73841: Pickle Jar does not make Pickles
Reporting client version: 514.1589

## Round ID:
[201400](https://scrubby.melonmesa.com/round/201400)

## Testmerges:
- [Nanotrasen Budget Programme - Mothball Edition [BIRDSHOT STATION]](https://www.github.com/tgstation/tgstation/pull/73502)
- [[NO GBP] Fixes some major (and minor) issues with the birthday station trait.](https://www.github.com/tgstation/tgstation/pull/73813)

## Reproduction:
1. Get the components for a pickle jar
2. Craft the pickle jar in the crafting menu
3. All the cucumbers used in the recipe remain as cucumbers, and do not become pickles

### PR DESCRIPTION
**Pull request by TheSmallBlue**
## About The Pull Request
Fixes #73841 

---

_It is the 12th of March, 2023. Around 3am. I have published a Pull Request which involves circuits, and got reminded of my low GBP. I go into the issues tab to see if there's anything someone of my low skill caliber could tackle. I see it; Pickles.
"How hard could I be?" I ask myself, foolishly unaware of the dangers that would soon overcome me.
Surely it must've been a mistype, I thought. Surely someone accidentally confused pickles and cucumbers.
"Wait, the pickles are supposed to be created on the jar when the jar is created", I say foolishly.
"Wait, its putting the ingredients used for the jar in the jar, that doesn't explain why the pickles aren't there though", I say foolishly
"Wait, whoever tried fixing this earlier fucking qdel'd the beaker and called it a day????", I say, foolishly._

---

Anyways I changed how the crafting menu distincts between categories, instead of checking whether or not the path is for food, it checks the actual categories themselves (why didn't it do this already), meaning that you can have non-food items on the food tab if it has a food category. Did this by adding a list that includes all crafting categories, so in the future when adding new categories you'll have to add them twice, which sucks, but oh well.

Also added a new variable to craftable items, which makes it so that you can not delete a container's contents if you so wish (why was this the default).

All this so that when you craft pickles, it actually crafts pickles instead of cucumbers.

I spent hours on this, its 6:30am as I'm typing this. I'm tired. Fucking pickles.

Super duper ultra thanks to FinalPotato for guiding me and suffering with me through this and teaching me so much about DM and BYOND. I cannot emphasize just how helpful and awesome they were thank you thank you thank you <3
## Why It's Good For The Game

Bug fixing be good
## Changelog
:cl:
fix: The jar of pickles, after millenia, finally actually contains pickles. All hail the jar of pickles.
/:cl:

### CODE REVIEW

File: code/__HELPERS/global_lists.dm [hunk 1]
```diff
@@ -65,8 +65,8 @@
 	for(var/path in subtypesof(/datum/crafting_recipe))
 		if(ispath(path, /datum/crafting_recipe/stack))
 			continue
-		var/is_cooking = ispath(path, /datum/crafting_recipe/food/)
 		var/datum/crafting_recipe/recipe = new path()
+		var/is_cooking = (recipe.category in GLOB.crafting_category_food)
```
Comment (SuperNovaa41) [inline code/__HELPERS/global_lists.dm:69]:
```suggestion
		var/is_cooking = recipe.category in GLOB.crafting_category_food
```

**Review by SuperNovaa41 — COMMENTED** [review]
small nitpick

**Comment by TheSmallBlue** [comment]
Yeah nope the parenthesees are there for a reason!

File: code/__HELPERS/global_lists.dm (hunk 1 above)
Comment (TheSmallBlue) [reply code/__HELPERS/global_lists.dm:69]: Not how that works, apparently!

**Pull request merged**